src/mongo/db/storage/input_stream.h

```diff
@@ -57,6 +58,11 @@ class InputStream {
      */
     virtual int readBytes(int count, char* buffer) = 0;
 
+    /**
+     * Collect occurrences of errors that occurred while reading the external inputStream
+     */
+    virtual ErrorCount getStats() const = 0;
```

Comment:
Probably we can define an interface for `InputStream`'s stats like...


```suggestion
struct InputStreamStats {
    virtual InputStreamStats* aggregate(const InputStreamStats* other) = 0;   
    virtual bool hasStats() const = 0;
    virtual BSONObj toBson() const = 0;
};
```

Reply:
Not exactly but Done through via different approach.

Reply:
I like your approach! Thanks for updating the PR!